pkg/ring/kv/dynamodb/dynamodb.go

```diff
@@ -59,7 +60,8 @@ func newDynamodbKV(cfg Config, logger log.Logger) (dynamodbKV, error) {
 
 	if len(cfg.Region) > 0 {
 		sess.Config = &aws.Config{
-			Region: aws.String(cfg.Region),
+			Region:     aws.String(cfg.Region),
+			HTTPClient: &http.Client{Timeout: cfg.HTTPClientTimeout},
```

Comment:
I am fine of adding this flag.
But is it better to implement a new `kv.Client` that adds context timeout for all operations? So that all Ring KV client can benefit from that.

Reply:
Thanks for the feedback, I'll go with @alanprot 's suggestion and add the timeout to the outer context instead to simplify it and ensure other clients can timeout on a hanging heartbeat